app/src/main/java/com/example/moviewithfragments/adapters/MovieRecyclerView.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.moviewithfragments.adapters
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide.with
+import com.example.moviewithfragments.R
+import com.example.moviewithfragments.models.MovieModel
+import com.example.moviewithfragments.utils.Credentials
+
+
+class MovieRecyclerView(private val onMovieListener: OnMovieListener) :
+    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
```

Comment:
Let's check `DiffUtil` because on every `notifyDataSetChanged()` it will refresh the whole list. we have to minimise it